### ISSUE TITLE
custom set prompt? Or netrepl may be confusing emacs on prompt recognition

### ISSUE THREAD

**Issue by Inc0n** [janet-lang/janet#1086]
TLDR: Janet repl works normally untill netrepl comes into the play.

Steps on to reproduce issue:

1. on any CMD line start janet with netrepl server using "-e" or in a script 
2. In EMACS connect to netrepl server using "janet -s -n -e <netrepl client sexp>"
    I have a fork of [velkyel/inf-janet](https://github.com/velkyel/inf-janet) which allows me to do so in one command
3. Now a process buffer connected via the netrepl is present 

The issue is netrepl has a prompt characteristic like: `[<ip>:<port>]:<nth prompt line>: "`.
This maybe considered as irregular for emacs. 
Here is my regex used for netrepl prompt: `"^[^\n]\\[[0-9.]+:[0-9]+\\]:[0-9]+: "` (comint-prompt-regex).
However, the process buffer in emacs will just be echoing the inputs. 
This would all work properly if connected to the netrepl in a terminal.

Workarounds I can think of, allow customize of prompt so it could be the same as normal janet repl.

**Comment by Inc0n** [comment]
Firstly, my gut feeling is that this issue is not related to prompt at all, more of the read of the repl examing spork library at [getline-async in netrepl/server](https://github.com/janet-lang/spork/blob/master/spork/netrepl.janet#L169).
Secondly, this issue is more relevant to spork as of now, would it be more appropriate to move the issue into that repo?

**Comment by bakpakin** [comment]
Much more relevant to spork (or the particular emacs packages). Sounds less like a bug and more like a problem with the emacs package - there are better ways to hook up to a repl than some regex hacks

**Comment by Inc0n** [comment]
> Much more relevant to spork

Thought so too.

> (or the particular emacs packages)

I guess this would be as far as I can get on my own. Well there is no particular support from any emacs package for janet with netrepl...

> Sounds less like a bug and more like a problem with the emacs package - there are better ways to hook up to a repl than some regex hacks

regex hacks? Hmm, fixing the `comint-prompt-regex` in emacs for inferior buffer interaction I thought would be a very standard thing to do, no?

Unless you are refereing to using `term` instead as the underlying support for process interaction? I just found julia-repl which does that.

**Comment by sogaiu** [comment]
Perhaps off-topic, but for future reference it appears that (the somewhat misnamed) https://github.com/sogaiu/a-janet-spork-client ended up working in this case.
